### DISCUSSION TITLE
Ontology generation

### DISCUSSION THREAD

**Discussion by swetepete** [explosion/spaCy#11790] [Help: Coding & Implementations]
I would like Spacy to generate an ontology, meaning that given textual data it creates its own labels as it identifies particular textual element types.

Libraries use subject classification systems where subjects can have sub-subjects, like China, China - History, and China - History - 13th Century, and an entity (a book) is classified with the most precise subject tag possible.

The goal is to identify any text that implies a “service” is being sought, any commercial service like mowing the lawn or setting up a computer or hiring a data scientist, but it can become quite specific like “Install new Mac OS” or something.

Spacy should generate the term for the service, they aren’t presupplied.

I am pretty sure this will require a pretrained transformer to have enough linguistic knowledge to create labels, creatively?

On a completely different note, is Spacy used for natural language generation?

**Comment by polm** [comment]
NLG is [specifically out of scope](https://spacy.io/usage/facts-figures#comparison-usage) for spaCy.
